src/components/DataSession/OperationPipeline.vue

```diff
@@ -56,20 +56,32 @@ function operationBtnColor(index) {
 async function pollOperationCompletion(operationID) {
   const url = store.datalabApiBaseUrl + 'datasessions/' + props.session_id + '/operations/' + operationID + '/'
 
-  const updateOperationPercentages = (response) => {
-    if (response && response.percent_completion !== undefined && response.status != 'COMPLETED') {
-      operationPercentages.value[operationID] = response.percent_completion * DEC_TO_PERCENT
-    } else {
-      operationPercentages.value[operationID] = COMPLETE_PERCENT
-      emit('operationCompleted', response)
-      if (operationID in operationPollingTimers) {
-        clearInterval(operationPollingTimers[operationID])
-        setTimeout(clearPollingData(operationID), POLL_WAIT_TIME)
+  const updateOperationStatus = (response) => {
```

Comment:
Is it possible to pass the operation id and the response so that it doesn't have to be nested?

Reply:
response is good anywhere since the callback will pass it to whatever function is there. 

Looking at operationID though it would be really tricky since its already passed as an argument to the parent function PollOperationCompletion, so to un-nest this function would mean adding a ref variable to track what pollOperationCompletion id is called.

[Good explanation for why certain functions make sense to nest ](https://stackoverflow.com/a/44482736/9627391) particularly the points 1, 2, and 5 for this case